core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Submission.java

```diff
@@ -25,72 +25,151 @@
 
 import java.io.Serializable;
 import java.util.Date;
-import java.util.Map;
+import java.util.HashSet;
 import java.util.Set;
 
-import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryDilution;
-import uk.ac.bbsrc.tgac.miso.core.data.type.SubmissionActionType;
-
-/**
- * Created by IntelliJ IDEA. User: davey Date: 10-Feb-2010 Time: 09:43:30
- */
-public interface Submission extends Nameable, Comparable<Submission>, Serializable {
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+@Entity
+@Table(name = "Submission")
+public class Submission implements Nameable, Comparable<Submission>, Serializable
+{
 
   public static final Long UNSAVED_ID = 0L;
 
-  public void setId(long id);
+  private static final long serialVersionUID = 1L;
+
+  private String accession;
+  private String alias;
+  private boolean completed;
+
+  @Temporal(TemporalType.DATE)
+  private Date creationDate;
+
+  private String description;
+
+  @ManyToMany(targetEntity = Experiment.class)
+  @JoinTable(name = "Submission_Experiment", joinColumns = {
+      @JoinColumn(name = "submission_submissionId") }, inverseJoinColumns = {
+          @JoinColumn(name = "experiments_experimentId") })
+  private Set<Experiment> experiments = new HashSet<>();
+
+  private String name;
 
-  public void setName(String name);
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long submissionId = UNSAVED_ID;
 
-  public String getAlias();
+  @Temporal(TemporalType.DATE)
+  private Date submittedDate;
 
-  public void setAlias(String alias);
+  private String title;
+  private boolean verified;
 
-  public String getAccession();
+  @Override
+  public int compareTo(Submission t) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1210)